src/lib/stepper/step-header.ts

```diff
@@ -101,4 +101,14 @@ export class MatStepHeader extends CdkStepHeader implements OnDestroy {
       optional: this.optional
     };
   }
+
+  _getDefaultIconForState(state: string): string {
+    if (state == 'number') {
+      return `${this.index + 1}`;
+    }
+    return ({
```

Comment:
Since this is called inside a data binding, we'll end up creating and throwing away a lot of objects. Consider using a couple of `if` and `return` instead.